File: Nian_Karl/QL_Nian/src/ql/ast/expression/Binary.java
package ql.ast.expression;

public abstract class Binary extends Expression {
	
	protected final Expression left;
	protected final Expression right;
	
	public Binary(Expression left, Expression right){
		this.left = left;
		this.right = right;
	}
	
	public Expression getLeft(){
		return this.left;
	}
	
	public Expression getRight(){
		return this.right;
	}
	
}

File: FelipezConde/QuestionnaireLanguage/AST/Nodes/Expression/Binary/Binary.cs
using AST.Nodes.Interfaces;
using AST.Representation;

namespace AST.Nodes.Expression.Binary
{
    public abstract class Binary : ASTNode, IExpression
    {
        private readonly IExpression left; 
        private readonly IExpression right;

        protected Binary(IExpression left, IExpression right, PositionInText pos)
            :base(pos)
        {
            this.left = left; 
            this.right = right;
        }

        public IExpression Left()
        { return left; }

        public IExpression Right()
        { return right; }

        public abstract T Accept<T>(ASTVisitors.IVisitor<T> visitor);
    }
}
